### DISCUSSION TITLE
vs 无法打开本项目xaml进行开发: 显示包无法进行更新、相关性或冲突验证。

### DISCUSSION THREAD

**Discussion by GD-Slime** [ywmoyue/biliuwp-lite#206] [General]
昨天写了一半的xaml, 今天突然就打不开了...
在打开xaml 设计器时有报错信息:
<details><summary>Details</summary>
<p>

```
System.Exception
包无法进行更新、相关性或冲突验证。

Windows 无法安装程序包app.aa13babf8-6330-4df2-8291-55a4aa8ac705_1.0.0.0_x86__8wekyb3d8bbwe，因为此程序包依赖于找不到的框架。提供由"CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"发布的框架"Microsoft.NET.CoreRuntime.2.2"，其中包含中性或x86处理器体系结构和最低版本2.2.31331.1，以及要安装的此包。
   在 Microsoft.VisualStudio.DesignTools.UwpDesignerHost.UwpHostPlatform.<>c__DisplayClass23_0.<EnsurePackageRegisteredAsync>b__0()
   在 System.Threading.Tasks.Task.Execute()
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.VisualStudio.DesignTools.DesignerHost.HostServices.HostPlatformBase.<CreateSurfaceProcessAsync>d__23.MoveNext()
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.VisualStudio.DesignTools.SurfaceDesigner.Documents.SurfaceIsolation.SurfaceProcessContext.<EnsureSurfaceProcessAsyncCore>d__108.MoveNext()
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 Microsoft.VisualStudio.DesignTools.SurfaceDesigner.Documents.SurfaceIsolation.SurfaceProcessContext.<EnsureSurfaceProcessAsyncInternal>d__106.MoveNext()
--- 引发异常的上一位置中堆栈跟踪的末尾 ---
   在 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   在 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   在 Microsoft.VisualStudio.DesignTools.XamlSurfaceDesigner.Views.IsolatedSurfaceImageHost.<CreateSurfaceAsync>d__67.MoveNext()

System.AggregateException
发生一个或多个错误。
```

</p>
</details> 

然而已经安装了这个框架, 压根没用:
![image](https://github.com/ywmoyue/biliuwp-lite/assets/82302542/8ae41f74-46c2-4a5c-b760-dc89fc1618f0)

猜测可能是这里缺框架了`C:\Program Files (x86)\Microsoft SDKs\Windows Kits\10\ExtensionSDKs\Microsoft.NET.CoreRuntime`:
![image](https://github.com/ywmoyue/biliuwp-lite/assets/82302542/b6b1d5d7-2d70-442a-845d-33dcf159f839)

然而找了全网也没有解决办法, 也没有这里需要的东西. vs卸载了重新安装也没效果. windows也退回了上个版本.
@ywmoyue 如果这里确实有2.2的文件的话 可否发一份给我😭 我先试试能不能行. 不行的话我再想其他方法.

**Comment by ywmoyue** [comment]
我这个路径下也没有2.2

我这里本地打包的debug包有Microsoft.NET.CoreRuntime.2.2.appx这个安装包 ，你可以试试在github的codespaces里面能不能运行msbuild把这个包跑出来

我这边看看怎么把这个包传上来

**Reply by ywmoyue** [reply]
[Microsoft.NET.CoreRuntime.2.2.appx.zip](https://github.com/ywmoyue/biliuwp-lite/files/12047919/Microsoft.NET.CoreRuntime.2.2.appx.zip)

**Reply by GD-Slime** [reply]
> [Microsoft.NET.CoreRuntime.2.2.appx.zip](https://github.com/ywmoyue/biliuwp-lite/files/12047919/Microsoft.NET.CoreRuntime.2.2.appx.zip)

安装了, 但还是一样的错误. 我在系统的设置-应用管理里也没有看到这个包. 真是太怪了.

**Comment by GD-Slime** [comment]
@ywmoyue 我在我的那个SDK目录里造了一个出来, 把你的包发进去, 然后就真的可以跳过这个错误了....
当然, 还是缺包, 不过起码证明这方法有效果. 
![image](https://github.com/ywmoyue/biliuwp-lite/assets/82302542/30428eca-0b5f-4226-ab7b-5d5c43f1de6b)

现在缺: Microsoft.NET.CoreFramework.Debug.2.2 这个包

**Reply by GD-Slime** [reply]
或者您可以直接把整个debug包传上来 或者我在codespace里打包? (不太会...)

**Reply by ywmoyue** [reply]
你看看有没有这个目录：C:\Program Files (x86)\Microsoft SDKs\UWPNuGetPackages\runtime.win10-x86.microsoft.net.uwpcoreruntimesdk

**Reply by ywmoyue** [reply]
可以试试用 Everything 进行文件搜索

**Reply by GD-Slime** [reply]
我找到并安装了这俩包, 然后新的错误又出现了:
![image](https://github.com/ywmoyue/biliuwp-lite/assets/82302542/3173be94-9d5b-4362-b0f2-4edce6284a9f)

有点命途多舛... 我备份一下数据, 过几天重置一下电脑吧.

**Comment by GD-Slime** [comment]
我重新安装了一下17763版本的Windows SDK, 问题解决了... 此贴终结.
